presto-main/src/main/java/com/facebook/presto/operator/scalar/JsonOperators.java

```diff
@@ -13,20 +13,153 @@
  */
 package com.facebook.presto.operator.scalar;
 
+import com.facebook.presto.metadata.OperatorType;
+import com.facebook.presto.spi.PrestoException;
 import com.facebook.presto.spi.type.StandardTypes;
 import com.facebook.presto.type.SqlType;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import io.airlift.slice.DynamicSliceOutput;
 import io.airlift.slice.Slice;
+import io.airlift.slice.SliceOutput;
+
+import java.io.IOException;
 
 import static com.facebook.presto.metadata.OperatorType.EQUAL;
 import static com.facebook.presto.metadata.OperatorType.HASH_CODE;
 import static com.facebook.presto.metadata.OperatorType.NOT_EQUAL;
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_CAST_ARGUMENT;
+import static com.facebook.presto.spi.type.StandardTypes.JSON;
+import static com.fasterxml.jackson.core.JsonFactory.Feature.CANONICALIZE_FIELD_NAMES;
+import static java.lang.String.format;
 
 public final class JsonOperators
 {
+    private static final JsonFactory JSON_FACTORY = new JsonFactory().disable(CANONICALIZE_FIELD_NAMES);
+
     private JsonOperators()
     {
     }
 
+    @ScalarOperator(OperatorType.CAST)
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice castToVarchar(@SqlType(StandardTypes.JSON) Slice slice)
+    {
+        return slice;
+    }
+
+    @ScalarOperator(OperatorType.CAST)
+    @SqlType(StandardTypes.BIGINT)
+    public static long castToBigint(@SqlType(StandardTypes.JSON) Slice json)
+    {
+        try {
+            try (JsonParser parser = JSON_FACTORY.createJsonParser(json.getInput())) {
+                parser.nextToken();
+                return parser.getLongValue();
```

Comment:
Check that there are no more tokens and throw `INVALID_CAST_ARGUMENT` if there are.